### PR TITLE
ui: Refine status indicators

### PR DESCRIPTION
**Pull request by jgwhite**
## Why the change?

0.4.0 introduced “status reports” (AKA “health checks”) to Waypoint. Status reports join the existing concepts of “state” and “status” on Waypoint entities. With the arrival of status reports, entities such as deployments have three complementary notions of “status”:

1. **Physical Status** status of underlying resources
2. **Operation Status** outcome of a waypoint operation (build, deploy, release)
3. **Health Status** outcome of most recent status report (AKA health check)

This PR aims to present all these statuses in an unambiguous manner, based on this proposed design:

image

Addresses #1581 and other issues with status indicators.

## What’s the plan?

- [x] Create `<StatusReportIndicator>` (to take over from `<StatusBadge>`)
- [x] Create `<OperationStatusIndicator>` (or similar)
- [x] Only show one “latest” card in each column
- [x] Update `<AppCard::Build>`
- [x] Update `<AppCard::Deployment>`
- [x] Update `<AppCard::Release>`
- [x] Update `<AppItem::Build>`
- [x] Update `<AppItem::Deployment>`
- [x] Update `<AppItem::Release>`
- [x] Update deployment detail page
- [x] Update release detail page
- [x] Retire `<StatusBadge>`
- [x] Await #1713 

## What’s out of scope?

- Artifact info

### Before

https://user-images.githubusercontent.com/34030/124609083-98217e00-de6f-11eb-8be2-cd6a4d77f264.mp4

### After

https://user-images.githubusercontent.com/34030/124751917-771c6400-df27-11eb-94b9-333cae1f13fa.mp4

https://user-images.githubusercontent.com/34030/124751922-77b4fa80-df27-11eb-9db6-276b710fa9fe.mp4

### Using Mirage

1. Check out the branch:
   ```sh
   git checkout ui/refine-status-indicators
   ```
2. Boot the dev server
   ```sh
   cd ui && ember serve
   ```
3. [Visit the app](http://localhost:4200/default/marketing-public/app/wp-matrix)
4. Verify you see the new status indicators
5. Verify the tooltips appear correctly
6. Navigate to builds, deployments, releases
7. Verify the new indicators look good wherever you see them

### For realsies

1. Check out the branch:
   ```sh
   git checkout ui/refine-status-indicators
   ```
2. Build the ember app
   ```sh
   (cd ui && make)
   ```
3. Bundle the static assets
   ```sh
   make static-assets
   ```
4. Build the dev server
   ```sh
   make docker/server
   ```
5. Build the CLI
   ```sh
   make bin
   ```
6. Install waypoint on your platform of choice, i.e.
   ```sh
   ./waypoint install -platform=kubernetes -accept-tos -k8s-server-image=waypoint:dev
   ```
7. Open the UI
   ```
   ./waypoint ui -authenticate
   ```
8. Try out an example app of your choice (i.e. [kubernetes/nodejs](https://github.com/hashicorp/waypoint-examples/tree/main/kubernetes/nodejs))
9. Verify you see the new status indicators

### CODE REVIEW

**Comment by almonk** [comment]
Love to see all the videos and context for this PR 👏

**Comment by almonk** [comment]
Refined the row design 👍

CleanShot 2021-07-07 at 11 22 45

CleanShot 2021-07-07 at 11 21 22

**Review by almonk — APPROVED** [review]

File: ui/app/components/app-card.hbs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      v5
+    </:meta-primary>
+    <:meta-secondary>
+      Starting…
+    </:meta-secondary>
+    <:actions>
+      <a href="...">Visit</a>
+    </:actions>
+  </AppCard>
+}}
```
Comment (gregone) [inline ui/app/components/app-card.hbs:15]: thanks for adding an example comment for the named blocks 👍  (Unrelated but is there a preferred convention for the casing for those?)

File: ui/app/styles/components/status-report-indicator.scss [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  &--alive &__badge {
+    background: color.$green-500;
+    animation: pulse 2s infinite;
+    @media (prefers-reduced-motion: reduce) {
+      animation: none;
+    }
+  }
+  &--alive &__label {
+    color: color.$green-700;
+    @media (prefers-color-scheme: dark) {
```
Comment (gregone) [inline ui/app/styles/components/status-report-indicator.scss:43]: This is great. We could simplify the colors by declaring them in variables in a `:root` pseudo-class up top, but frankly this is already fine and readable

**Review by gregone — APPROVED** [review]
✨ :shipit: Excellent!

File: ui/app/components/app-card.hbs (hunk 1 above)
Comment (jgwhite) [reply ui/app/components/app-card.hbs:15]: That’s a good question. [The RFC](https://github.com/emberjs/rfcs/blob/master/text/0460-yieldable-named-blocks.md) uses lowercase names for the blocks, but doesn’t specify anything strict.

File: ui/app/styles/components/status-report-indicator.scss (hunk 2 above)
Comment (jgwhite) [reply ui/app/styles/components/status-report-indicator.scss:43]: Thanks!

I tried to reuse as many of the `:root` variables as I could keep straight in my head. There are probably more opportunities to consolidate but I think this does the trick for now.

**Pull request merged**